praxis-isncsci/ui
# Color tokens

Closes https://github.com/praxis-isncsci/ui/issues/9

## Problem

**Figma** variables are our _source of truth_. They are exported into a `json` file using the plugin **Variables import export**. We need an automated way to import the tokens generated into our project.

## Solution

This PR adds the package `style-dictionary` to our dev dependencies so we can convert the file generated from **Figma** into _CSS_ files by running `npm run update-design-tokens`. The script `scripts/generate-token-files.ts` has the code that performs the convertion.

The documentation on how the color palettes are generated, as well as the colors themselves, is in the **Storybook** story `Designsystem/color/docs`.

## Testing

- [ ] 1. Can generate new styles from `json` file

  <details>
    <summary>Use case</summary>

    1. `npm run storybook`
    2. In *Storybook*, navigate to http://localhost:6006/?path=/docs/designsystem-color--docs
    3. In the project, open the file `assets/tokens/tokens.json` - This file is generated from **Figma**
    4. Add a new color with a few variants
    5. `npm run update-design-tokens`
    6. Refresh **Storybook**
    7. Confirm the new colors were added to the palette

  </details>

---
